rainier-core/src/main/scala/com/stripe/rainier/core/Continuous.scala

```diff
@@ -32,7 +27,7 @@ trait LocationScaleFamily { self =>
       Generator.from { (r, n) =>
         generate(r)
       }
-    def realLogDensity(real: Real): Real =
+    def logDensity(real: Real): Real =
```

Comment:
What's the advantage of this `self` pattern over just making this an abstract `trait` with an unimplemented `logDensity` method? When I read about this pattern it seemed to be used when you wanted to specify a type on the `self` parameter.

Reply:
This is a bit strange - the outer `LocationScaleFamily` here is a trait for, basically, factory objects that produce new instances of `Continuous`. So in this case the `self` is just so that the definition of `logDensity` inside the definition of this new `Continuous` object can refer to the outer factory object (whereas `this` would refer to the `Continuous` object).

Reply:
Oh, I see. Cool!